From 3c7673320dbc70197230e3ebc977be69d1c35870 Mon Sep 17 00:00:00 2001
Message-Id: <3c7673320dbc70197230e3ebc977be69d1c35870.1524081065.git.Jim.Somerville@windriver.com>
In-Reply-To: <8d084dcc717666af0283e4ddeefaba7e331d879a.1524081065.git.Jim.Somerville@windriver.com>
References: <8d084dcc717666af0283e4ddeefaba7e331d879a.1524081065.git.Jim.Somerville@windriver.com>
From: Alex Kozyrev <alex.kozyrev@windriver.com>
Date: Mon, 11 Sep 2017 17:05:22 -0400
Subject: [PATCH 28/39] fix high latency reported by cyclictest

CGTS-7179 reported cyclictest latency spikes on low-latency compute profile.
The issue was injected into TC during kernel rebase to 3.10.0-514.16.1.el7.
Latency has been hurt by addition of the timer_migration sysctl.
To cure the issue we need to port back two patches from latest stable kernel:
bc7a34b8b9ebfb0f4b8a35a72a0b134fd6c5ef50
timer: Reduce timer migration overhead if disabled
683be13a284720205228e29207ef11a1c3c322b9
timer: Minimize nohz off overhead
Little modification were required to apply these patches nicely.

Signed-off-by: Alex Kozyrev <alex.kozyrev@windriver.com>
Signed-off-by: Jim Somerville <Jim.Somerville@windriver.com>
---
 SPECS/kernel-rt.spec | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/SPECS/kernel-rt.spec b/SPECS/kernel-rt.spec
index ca8fb19..69ba548 100644
--- a/SPECS/kernel-rt.spec
+++ b/SPECS/kernel-rt.spec
@@ -427,6 +427,8 @@ Patch1032: nfsd-check-for-oversized-NFSv2-v3-arguments.patch
 Patch1033: nfsd4-minor-NFSv2-v3-write-decoding-cleanup.patch
 Patch1034: nfsd-stricter-decoding-of-write-like-NFSv2-v3-ops.patch
 Patch1035: US101216-IMA-support-in-Titanium-kernel.patch
+Patch1036: timer-reduce-timer-migration-overhead-if-disabled.patch
+Patch1037: timer-minimize-nohz-off-overhead.patch
 
 BuildRoot: %{_tmppath}/kernel-%{KVERREL}-root
 
@@ -822,6 +824,8 @@ ApplyPatch nfsd-check-for-oversized-NFSv2-v3-arguments.patch
 ApplyPatch nfsd4-minor-NFSv2-v3-write-decoding-cleanup.patch
 ApplyPatch nfsd-stricter-decoding-of-write-like-NFSv2-v3-ops.patch
 ApplyPatch US101216-IMA-support-in-Titanium-kernel.patch
+ApplyPatch timer-reduce-timer-migration-overhead-if-disabled.patch
+ApplyPatch timer-minimize-nohz-off-overhead.patch
 
 # move off upstream version mechanism
 if [ -e localversion-rt ]; then
-- 
1.8.3.1

